### PR TITLE
Persist URI to preference after choosing shared folder path

### LINKED ISSUE #2626: FTP MLSD command failed
**Describe the bug**

Working telnet session (so the required FTP commands are sure to be entered):
```
airwave@hsv010 ~ $ telnet 192.168.191.187 2211
Trying 192.168.191.187...
Connected to 192.168.191.187.
Escape character is '^]'.
220 Service ready for new user.
USER anonymous
331 Guest login okay, send your complete e-mail address as password.
PASS test@test.com
230 User logged in, proceed.
SYST
215 UNIX Type: Apache FtpServer
SITE
200 Command SITE okay. Use SITE HELP to get more information.
FEAT
211-Extensions supported
 SIZE
 MDTM
 REST STREAM
 LANG en;zh-tw;ja;is
 MLST Size;Modify;Type;Perm
 AUTH SSL
 AUTH TLS
 MODE Z
 UTF8
 TVFS
 MD5
 MMD5
 MFMT
211 End
PASV
227 Entering Passive Mode (192,168,191,187,133,158)
MLSD
150 File status okay; about to open data connection.
226 Closing data connection.
QUIT
221 Goodbye.
Connection closed by foreign host.
```

Not working telnet session:
```
airwave@hsv010 ~ $ telnet 192.168.191.187 2211
Trying 192.168.191.187...
Connected to 192.168.191.187.
Escape character is '^]'.
220 Service ready for new user.
USER anonymous
331 Guest login okay, send your complete e-mail address as password.
PASS test@test.com
550 Requested action not taken.
FEAT
211-Extensions supported
 SIZE
 MDTM
 REST STREAM
 LANG en;zh-tw;ja;is
 MLST Size;Modify;Type;Perm
 AUTH SSL
 AUTH TLS
 MODE Z
 UTF8
 TVFS
 MD5
 MMD5
 MFMT
211 End
PASV
227 Entering Passive Mode (192,168,191,187,233,137)
MLSD
150 File status okay; about to open data connection.
550 Requested action not taken.
QUIT
221 Goodbye.
Connection closed by foreign host.
```

Stacktrace:
```
2021-06-14 12:44:40.789 9631-9738/com.amaze.filemanager W/DefaultFtpHandler: RequestHandler.service()
    java.lang.NullPointerException: Attempt to invoke interface method 'org.apache.ftpserver.ftplet.FtpFile org.apache.ftpserver.ftplet.FileSystemView.getFile(java.lang.String)' on a null object reference
        at org.apache.ftpserver.command.impl.listing.DirectoryLister.listFiles(DirectoryLister.java:97)
        at org.apache.ftpserver.command.impl.listing.DirectoryLister.listFiles(DirectoryLister.java:75)
        at org.apache.ftpserver.command.impl.MLSD.execute(MLSD.java:114)
        at org.apache.ftpserver.impl.DefaultFtpHandler.messageReceived(DefaultFtpHandler.java:211)
        at org.apache.ftpserver.listener.nio.FtpHandlerAdapter.messageReceived(FtpHandlerAdapter.java:62)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain$TailFilter.messageReceived(DefaultIoFilterChain.java:858)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:542)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1300(DefaultIoFilterChain.java:48)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:947)
        at org.apache.ftpserver.listener.nio.FtpLoggingFilter.messageReceived(FtpLoggingFilter.java:85)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:542)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1300(DefaultIoFilterChain.java:48)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:947)
        at org.apache.mina.core.filterchain.IoFilterEvent.fire(IoFilterEvent.java:74)
        at org.apache.mina.filter.logging.MdcInjectionFilter.filter(MdcInjectionFilter.java:135)
        at org.apache.mina.filter.util.CommonEventFilter.messageReceived(CommonEventFilter.java:70)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:542)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1300(DefaultIoFilterChain.java:48)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:947)
        at org.apache.mina.filter.codec.ProtocolCodecFilter$ProtocolDecoderOutputImpl.flush(ProtocolCodecFilter.java:398)
        at org.apache.mina.filter.codec.ProtocolCodecFilter.messageReceived(ProtocolCodecFilter.java:234)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:542)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1300(DefaultIoFilterChain.java:48)
        at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:947)
        at org.apache.mina.core.filterchain.IoFilterEvent.fire(IoFilterEvent.java:74)
        at org.apache.mina.core.session.IoEvent.run(IoEvent.java:63)
        at org.apache.mina.filter.executor.OrderedThreadPoolExecutor$Worker.runTask(OrderedThreadPoolExecutor.java:770)
        at org.apache.mina.filter.executor.OrderedThreadPoolExecutor$Worker.runTasks(OrderedThreadPoolExecutor.java:762)
        at org.apache.mina.filter.executor.OrderedThreadPoolExecutor$Worker.run(OrderedThreadPoolExecutor.java:704)
        at java.lang.Thread.run(Thread.java:919)
2021-06-14 12:44:40.791 9631-9738/com.amaze.filemanager I/FtpLoggingFilter: SENT: 150 File status okay; about to open data connection.
2021-06-14 12:44:40.792 9631-9738/com.amaze.filemanager I/FtpLoggingFilter: SENT: 550 Requested action not taken.
```

**To Reproduce**
1. Install Amaze 3.5.3
2. Start FTP server
3. Use FTP client that supports `MLSD` command at first place, or use `telnet` to enter the commands directly, it's working
4. Upgrade to Amaze 3.6
5. Start FTP server
6. Seeing grant URI permission dialog. Tap OK, opens DocumentsUI and tap allow access
7. FTP server starts
8. Use FTP client that supports `MLSD` command at first place, or use `telnet` to enter the commands directly
9. Seeing error. Consequent logins were also failed

**Expected behavior**
The two FTP sessions should look exactly the same, i.e. the MLSD command should continue to work after the upgrade.

**Smartphone (please complete the following information):**
 - Device: Oneplus 2
 - OS: /e/ OS 0.16 (Android 10)
 - Rooted: Yes
 - Version: 3.5.3 -> 3.6.0

**Additional context**
Possibily the URI grant action didn't save the URI returned from `ACTION_OPEN_DOCUMENT_TREE`, causing the new `DocumentFile`-based implementation of FTP server stopped working at this point.

### PR DESCRIPTION
**Pull request by TranceLove**
## Description
Continuation of #2598, fixes #2626. Saves the selected folder returned from ACTION_OPEN_DOCUMENT_TREE Intent.

#### Issue tracker   
Fixes #2626

#### Manual tests
- [x] Done  
  
- Device: Oneplus 2
- OS: /e/ OS 0.16 (Android 10)

#### Build tasks success  
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by VishalNehra** [comment]
Build test apk

File: app/src/main/java/com/amaze/filemanager/ui/fragments/FtpServerFragment.kt [hunk 1]
```diff
@@ -407,13 +410,14 @@ class FtpServerFragment : Fragment(R.layout.fragment_ftp) {
                                 activityResultHandlerOnFtpServerPathGrantedSafAccess.launch(
                                     Intent(Intent.ACTION_OPEN_DOCUMENT_TREE).also {
                                         if (Build.VERSION.SDK_INT >= O &&
-                                            directoryUri.equals(DEFAULT_PATH)
+                                            directoryUri!!.startsWith(DEFAULT_PATH)
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/ui/fragments/FtpServerFragment.kt:413]: No `!!` please

Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/ui/fragments/FtpServerFragment.kt]: I too know that would be silly... but with the default value given to `SharedPreferences.getString(key, defaultValue)` is a non-null, it's safe to assume `directoryUri` is definitely a non-null.

So there will still need a `!!` anyway. But I'm putting that up to the `SharedPreferences.getString()` call now.

**Review by EmmanuelMess — APPROVED** [review]
Tested on LG K11alpha

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**